src/nvim/screen.c

```diff
@@ -1570,6 +1570,17 @@ static void win_update(win_T *wp)
     got_int = save_got_int;
 }
 
+///
+/// Returns maximum length of the signcolumn
+///
+/// @param wp window we want signcolumn width from
+///
+/// @return max width of signcolumn (cell unit)
+int win_get_colwidth(win_T *wp)
+{
+  // 2 is the default value
+  return 2;
```

Comment:
It deals only with the signcolumn and returns the width in cell unit that is used for display (I suppose you could call it a maximum).
Obviously now it's weird to have just a "return 2" but I intend to have it return something that adapts to the maximum (cell) width of signs on a line (I read several times on the gitter it was a concern that neomake signs override vim-signature marks for instance; it bothers me as well).
I suppose such a value would be cached in the buffer structure hence 'buf_signcol_max' might be preferable over 'win_signcol_max' ?
I could write a proof of concept if you want. I've read quickly the mark/sign code it seems that vim just looks for the first sign so getting the maximum width could involve API change, hence I limited myself to this simple commit first.